### PR TITLE
use Sam getUserIds api inside free credits code [risk: low]

### PR DESCRIPTION
**Pull request by davidangb**
The free credits termination flow is ignorant of -  and fails for - users who have different subjectids in Sam and Google, due to those users arriving via invite. This PR addresses that case.

See DataBiosphere/firecloud-app#284.

Because this error was found by Onix during the termination process, I have confidence that invited users can progress correctly through enablement and enrollment  for free credits.  It's only once they hit termination that we are seeing a problem. The termination process relied on Sam's admin API to look up user's subjectid based on their email. This returned the Sam-generated  "invite id", while what we actually needed is the user's "google id" in order to query  their state in thurloe.

-----

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: script/build.sh [hunk 1]
```diff
@@ -136,7 +136,7 @@ function cleanup()
 {
     echo "cleaning up..."
     if [[ -n $SERVICE_ACCT_KEY_FILE ]]; then
-      gcloud auth revoke
+      gcloud auth revoke && echo 'Token revoke succeeded' || echo 'Token revoke failed -- skipping'
```
Comment (davidangb) [inline script/build.sh:139]: a la broadinstitute/sam#287

**Comment by davidangb** [comment]
```
[info] Tests: succeeded 127, failed 1, canceled 0, ignored 9, pending 0
[info] *** 1 TEST FAILED ***
[error] Failed tests:
[error] 	org.broadinstitute.dsde.firecloud.test.billing.BillingSpec
[error] (Test / testOnly) sbt.TestsFailedException: Tests unsuccessful
```
_"A user with a billing account with a new billing project should be able add a new user, create a workspace, and run a method, change billing account"_ failed - we know this one fails  every so often. I'm putting this PR up for review and will manually  re-run that spec.

File: src/test/scala/org/broadinstitute/dsde/firecloud/webservice/TrialApiServiceSpec.scala [hunk 2]
```diff
@@ -737,5 +768,7 @@ object TrialApiServiceSpec {
     enrolledUser -> enrolledUserRegInfo,
     terminatedUser -> terminatedUserRegInfo,
     finalizedUser -> finalizedUserRegInfo,
-    registeredUser -> registeredUserRegInfo)
+    registeredUser -> registeredUserRegInfo,
+    invitedUser -> invitedUserRegInfo
+  )
```
Comment (davidangb) [inline src/test/scala/org/broadinstitute/dsde/firecloud/webservice/TrialApiServiceSpec.scala:773]: this entire spec is a bit messy - the mock data setup and mock dao logic are rather scattered. I'm actively trying to not refactor the whole thing to avoid scope creep.

**Review by dvoet — APPROVED** [review]

**Comment by davidangb** [comment]
`BillingSpec` passed in a manual run: https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/17834/
